### PR TITLE
create image service

### PR DESCRIPTION
**Pull request by Cheephot**

### CODE REVIEW

File: src/ImageService.Models.Db/DbImagesProject.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using System;
+
+namespace LT.DigitalOffice.ImageService.Models.Db
+{
+    public class DbImagesProject
+    {
+        public const string TableName = "ImagesProject";
+
+        public Guid Id { get; set; }
```
Comment (VLiamin) [inline src/ImageService.Models.Db/DbImagesProject.cs:11]: tabs

File: src/ImageService.Models.Db/DbImagesUser.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+using System;
+
+namespace LT.DigitalOffice.ImageService.Models.Db
+{
+    public class DbImagesUser
+    {
+        public const string TableName = "ImagesUser";
+
+        public Guid Id { get; set; }
+		public Guid? ParentId { get; set; }
```
Comment (VLiamin) [inline src/ImageService.Models.Db/DbImagesUser.cs:12]: tabs

**Review by VLiamin — APPROVED** [review]

File: src/ImageService.Models.Db/DbImagesNews.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        public void Configure(EntityTypeBuilder<DbImagesNews> builder)
+        {
+            builder
+                .ToTable(DbImagesNews.TableName);
+
+            builder
+                .HasKey(a => a.Id);
+
+            builder
+                .Property(a => a.Name)
```
Comment (Glunky) [inline src/ImageService.Models.Db/DbImagesNews.cs:31]: why name is required?

File: src/ImageService/Startup.cs [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+            });
+
+            services.AddMassTransitHostedService();
+        }
+
+        private void ConfigureEndpoints(
+            IBusRegistrationContext context,
+            IRabbitMqBusFactoryConfigurator cfg)
+        {
+            /*cfg.ReceiveEndpoint(_rabbitMqConfig.NameEndPoint, ep =>
```
Comment (Glunky) [inline src/ImageService/Startup.cs:163]: remove comments

**Review by Glunky — APPROVED** [review]

File: src/ImageService/Startup.cs (hunk 4 above)
Comment (Cheephot) [reply src/ImageService/Startup.cs:163]: Left comments so far for an example

File: src/ImageService.Models.Db/DbImagesNews.cs (hunk 3 above)
Comment (Cheephot) [reply src/ImageService.Models.Db/DbImagesNews.cs:31]: This error, fixed

File: src/ImageService.Models.Db/DbImagesProject.cs (hunk 1 above)
Comment (Cheephot) [reply src/ImageService.Models.Db/DbImagesProject.cs:11]: done

File: src/ImageService.Models.Db/DbImagesUser.cs (hunk 2 above)
Comment (Cheephot) [reply src/ImageService.Models.Db/DbImagesUser.cs:12]: done

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        protected override void Up(MigrationBuilder builder)
+        {
+            builder.CreateTable(
+                name: DbImagesUser.TableName,
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(nullable: false),
+                    ParentId = table.Column<Guid>(nullable: true),
+                    Name = table.Column<string>(nullable: true),
+                    Content = table.Column<Guid>(nullable: true),
```
Comment (BelehovEgor) [inline src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs:22]: i guess wrong type

File: src/ImageService.Models.Db/DbImagesUser.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+		public string Name { get; set; }
+		public string Content { get; set; }
+		public string Extentions { get; set; }
+		public DateTime CreatedAtUTC { get; set; }
```
Comment (BelehovEgor) [inline src/ImageService.Models.Db/DbImagesUser.cs:16]: change Utc

File: src/ImageService/Startup.cs [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+
+            _serviceInfoConfig = Configuration
+                .GetSection(BaseServiceInfoConfig.SectionName)
+                .Get<BaseServiceInfoConfig>();
+
+            _rabbitMqConfig = Configuration
+                .GetSection(BaseRabbitMqConfig.SectionName)
+                .Get<RabbitMqConfig>();
+
+            Version = "???";
```
Comment (BelehovEgor) [inline src/ImageService/Startup.cs:46]: 1 0 0 0

File: src/ImageService/Startup.cs [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+
+        #endregion
+
+        #region private methods
+
+        private void ConfigureMassTransit(IServiceCollection services)
+        {
+            services.AddMassTransit(x =>
+            {
+                //x.AddConsumer<NameConsumer>();
```
Comment (BelehovEgor) [inline src/ImageService/Startup.cs:140]: what is this?

File: src/ImageService/Startup.cs (hunk 4 above)
Comment (BelehovEgor) [reply src/ImageService/Startup.cs:163]: remove

File: src/ImageService/appsettings.json [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    // ProjectService
+
+    // MessageService
+
+    // ImageService
+
+  },
+  "CheckTokenMiddleware": {
+    "SkippedEndpoints": [
+      "/password/forgot",
```
Comment (BelehovEgor) [inline src/ImageService/appsettings.json:51]: delete

**Review by BelehovEgor — DISMISSED** [review]

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs [hunk 10]
```diff
@@ -0,0 +1,84 @@
+﻿using LT.DigitalOffice.ImageService.Models.Db;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Migrations;
+using System;
+
```
Comment (MariyaSin) [inline src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs:5]: remove line

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+                });
+
+            builder.CreateTable(
+                name: DbImagesNews.TableName,
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(nullable: false),
+                    ParentId = table.Column<Guid>(nullable: true),
+                    Name = table.Column<string>(nullable: true),
+                    Content = table.Column<Guid>(nullable: true),
```
Comment (MariyaSin) [inline src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs:39]: false

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+        {
+            builder.CreateTable(
+                name: DbImagesUser.TableName,
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(nullable: false),
+                    ParentId = table.Column<Guid>(nullable: true),
+                    Name = table.Column<string>(nullable: true),
+                    Content = table.Column<Guid>(nullable: true),
+                    Extentions = table.Column<bool>(nullable: false),
```
Comment (MariyaSin) [inline src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs:23]: rename to Extension

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+
+            builder.CreateTable(
+                name: DbImagesNews.TableName,
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(nullable: false),
+                    ParentId = table.Column<Guid>(nullable: true),
+                    Name = table.Column<string>(nullable: true),
+                    Content = table.Column<Guid>(nullable: true),
+                    Extentions = table.Column<bool>(nullable: false),
```
Comment (MariyaSin) [inline src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs:40]: rename to Extension

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+
+            builder.CreateTable(
+                name: DbImagesMessage.TableName,
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(nullable: false),
+                    ParentId = table.Column<Guid>(nullable: true),
+                    Name = table.Column<string>(nullable: true),
+                    Content = table.Column<Guid>(nullable: true),
+                    Extentions = table.Column<bool>(nullable: false),
```
Comment (MariyaSin) [inline src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs:57]: rename to Extension

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+
+            builder.CreateTable(
+                name: DbImagesProject.TableName,
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(nullable: false),
+                    ParentId = table.Column<Guid>(nullable: true),
+                    Name = table.Column<string>(nullable: true),
+                    Content = table.Column<Guid>(nullable: true),
+                    Extentions = table.Column<bool>(nullable: false),
```
Comment (MariyaSin) [inline src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs:74]: rename to Extension

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs (hunk 5 above)
Comment (MariyaSin) [reply src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs:22]: nullable: false

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+                });
+
+            builder.CreateTable(
+                name: DbImagesMessage.TableName,
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(nullable: false),
+                    ParentId = table.Column<Guid>(nullable: true),
+                    Name = table.Column<string>(nullable: true),
+                    Content = table.Column<Guid>(nullable: true),
```
Comment (MariyaSin) [inline src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs:56]: typw string, nullable false

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+                });
+
+            builder.CreateTable(
+                name: DbImagesProject.TableName,
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(nullable: false),
+                    ParentId = table.Column<Guid>(nullable: true),
+                    Name = table.Column<string>(nullable: true),
+                    Content = table.Column<Guid>(nullable: true),
```
Comment (MariyaSin) [inline src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs:73]: type string, nullable false

File: src/ImageService.Models.Db/DbImagesMessage.cs [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+	{
+        public const string TableName = "ImagesMessage";
+
+        public Guid Id { get; set; }
+		public Guid? ParentId { get; set; }
+		public string Name { get; set; }
+		public string Content { get; set; }
+		public string Extentions { get; set; }
+		public DateTime CreatedAtUTC { get; set; }
+		public Guid CreatedBy { get; set; }
```
Comment (MariyaSin) [inline src/ImageService.Models.Db/DbImagesMessage.cs:17]: remove unnecessary tab

File: src/ImageService.Models.Db/DbImagesMessage.cs [hunk 19]
```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using System;
+
+namespace LT.DigitalOffice.ImageService.Models.Db
+{
+    public class DbImagesMessage
+	{
+        public const string TableName = "ImagesMessage";
```
Comment (MariyaSin) [inline src/ImageService.Models.Db/DbImagesMessage.cs:9]: the table name must be plural. Add "s" to all tables in the end of the word

File: src/ImageService.Models.Db/DbImagesMessage.cs [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+
+        public Guid Id { get; set; }
+		public Guid? ParentId { get; set; }
+		public string Name { get; set; }
+		public string Content { get; set; }
+		public string Extentions { get; set; }
```
Comment (MariyaSin) [inline src/ImageService.Models.Db/DbImagesMessage.cs:15]: rename to Extension

File: src/ImageService.Models.Db/DbImagesNews.cs [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+{
+    public class DbImagesNews
+    {
+        public const string TableName = "ImagesNews";
+
+        public Guid Id { get; set; }
+        public Guid? ParentId { get; set; }
+        public string Name { get; set; }
+        public string Content { get; set; }
+        public string Extentions { get; set; }
```
Comment (MariyaSin) [inline src/ImageService.Models.Db/DbImagesNews.cs:15]: rename to Extension

File: src/ImageService.Models.Db/DbImagesProject.cs [hunk 22]
```diff
@@ -0,0 +1,39 @@
...
+		public Guid? ParentId { get; set; }
+		public string Name { get; set; }
+		public string Content { get; set; }
+		public string Extentions { get; set; }
```
Comment (MariyaSin) [inline src/ImageService.Models.Db/DbImagesProject.cs:15]: rename to Extension

File: src/ImageService.Models.Db/DbImagesUser.cs [hunk 23]
```diff
@@ -0,0 +1,39 @@
...
+{
+    public class DbImagesUser
+    {
+        public const string TableName = "ImagesUser";
+
+        public Guid Id { get; set; }
+		public Guid? ParentId { get; set; }
+		public string Name { get; set; }
+		public string Content { get; set; }
+		public string Extentions { get; set; }
```
Comment (MariyaSin) [inline src/ImageService.Models.Db/DbImagesUser.cs:15]: rename to Extension

File: src/ImageService.Models.Db/DbImagesProject.cs [hunk 24]
```diff
@@ -0,0 +1,39 @@
...
+		public DateTime CreatedAtUTC { get; set; }
```
Comment (MariyaSin) [inline src/ImageService.Models.Db/DbImagesProject.cs:16]: rename to CreatedAtUtc

File: src/ImageService.Models.Db/DbImagesNews.cs [hunk 25]
```diff
@@ -0,0 +1,39 @@
...
+        public DateTime CreatedAtUTC { get; set; }
```
Comment (MariyaSin) [inline src/ImageService.Models.Db/DbImagesNews.cs:16]: CreatedAtUtc

File: src/ImageService.Models.Db/DbImagesMessage.cs [hunk 26]
```diff
@@ -0,0 +1,39 @@
...
+		public DateTime CreatedAtUTC { get; set; }
```
Comment (MariyaSin) [inline src/ImageService.Models.Db/DbImagesMessage.cs:16]: CreatedAtUtc

File: src/ImageService.Models.Db/DbImagesMessage.cs (hunk 19 above)
Comment (Cheephot) [reply src/ImageService.Models.Db/DbImagesMessage.cs:9]: done

File: src/ImageService.Models.Db/DbImagesMessage.cs (hunk 26 above)
Comment (Cheephot) [reply src/ImageService.Models.Db/DbImagesMessage.cs:16]: done

File: src/ImageService.Models.Db/DbImagesNews.cs (hunk 21 above)
Comment (Cheephot) [reply src/ImageService.Models.Db/DbImagesNews.cs:15]: done

File: src/ImageService.Models.Db/DbImagesNews.cs (hunk 25 above)
Comment (Cheephot) [reply src/ImageService.Models.Db/DbImagesNews.cs:16]: done

File: src/ImageService.Models.Db/DbImagesProject.cs (hunk 22 above)
Comment (Cheephot) [reply src/ImageService.Models.Db/DbImagesProject.cs:15]: done

File: src/ImageService.Models.Db/DbImagesProject.cs (hunk 24 above)
Comment (Cheephot) [reply src/ImageService.Models.Db/DbImagesProject.cs:16]: done

File: src/ImageService.Models.Db/DbImagesUser.cs (hunk 23 above)
Comment (Cheephot) [reply src/ImageService.Models.Db/DbImagesUser.cs:15]: done

File: src/ImageService.Models.Db/DbImagesUser.cs (hunk 6 above)
Comment (Cheephot) [reply src/ImageService.Models.Db/DbImagesUser.cs:16]: done

File: src/ImageService/Startup.cs (hunk 7 above)
Comment (Cheephot) [reply src/ImageService/Startup.cs:46]: done

File: src/ImageService/Startup.cs (hunk 8 above)
Comment (Cheephot) [reply src/ImageService/Startup.cs:140]: Comment for example

Comment (Cheephot) [reply src/ImageService/Startup.cs:140]: I remove

File: src/ImageService/appsettings.json (hunk 9 above)
Comment (Cheephot) [reply src/ImageService/appsettings.json:51]: done

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs (hunk 5 above)
Comment (Cheephot) [reply src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs:22]: done

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs (hunk 15 above)
Comment (Cheephot) [reply src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs:74]: done

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs (hunk 10 above)
Comment (Cheephot) [reply src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs:5]: done

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs [hunk 27]
```diff
@@ -35,9 +34,9 @@ protected override void Up(MigrationBuilder builder)
                 {
                     Id = table.Column<Guid>(nullable: false),
                     ParentId = table.Column<Guid>(nullable: true),
-                    Name = table.Column<string>(nullable: true),
+                    Name = table.Column<string>(nullable: false),
                     Content = table.Column<Guid>(nullable: true),
```
Comment (ndoriomedov) [inline src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs:38]: content is string, nullable is false
name - nullable is true

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs [hunk 28]
```diff
@@ -0,0 +1,83 @@
...
+                    table.PrimaryKey("PK_ImagesNews", x => x.Id);
+                });
+
+            builder.CreateTable(
+                name: DbImagesMessage.TableName,
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(nullable: false),
+                    ParentId = table.Column<Guid>(nullable: true),
+                    Name = table.Column<string>(nullable: false),
```
Comment (ndoriomedov) [inline src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs:54]: name nullable: true

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs [hunk 29]
```diff
@@ -0,0 +1,83 @@
...
+    {
+        protected override void Up(MigrationBuilder builder)
+        {
+            builder.CreateTable(
+                name: DbImagesUser.TableName,
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(nullable: false),
+                    ParentId = table.Column<Guid>(nullable: true),
+                    Name = table.Column<string>(nullable: false),
```
Comment (MariyaSin) [inline src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs:20]: nullable true

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs [hunk 30]
```diff
@@ -0,0 +1,83 @@
...
+                });
+
+            builder.CreateTable(
+                name: DbImagesMessage.TableName,
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(nullable: false),
+                    ParentId = table.Column<Guid>(nullable: true),
+                    Name = table.Column<string>(nullable: false),
+                    Content = table.Column<Guid>(nullable: true),
```
Comment (ndoriomedov) [inline src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs:55]: content type is string, nullable - false

Comment (Cheephot) [reply src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs]: done

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs (hunk 28 above)
Comment (Cheephot) [reply src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs]: done

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs (hunk 27 above)
Comment (Cheephot) [reply src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs]: done

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs (hunk 29 above)
Comment (Cheephot) [reply src/ImageService.Data.Provider.MsSql.Ef/Migrations/20210820233300_InitialTable.cs]: done

**Review by MariyaSin — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**